edb/common/ast/match.py

```diff
@@ -1,206 +0,0 @@
-#
-# This source file is part of the EdgeDB open source project.
-#
-# Copyright 2010-present MagicStack Inc. and the EdgeDB authors.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-
-"""Generic AST tree pattern matching."""
-
-from __future__ import annotations
-
-import collections.abc
-import types
-
-from edb.common import adapter
-
-
-class MatchASTMeta(adapter.Adapter):
-    pass
```

Comment:
@msullivan your dept toward our remove-meta-classes fund has been erased.

Reply:
I removed a bunch of metaclasses from the parser a while back so I think he's in the surplus now and should re-add some more for balance.